modules/javafx.controls/src/main/java/javafx/scene/control/skin/ChoiceBoxSkin.java

```diff
@@ -206,6 +206,9 @@ public ChoiceBoxSkin(ChoiceBox<T> control) {
 
     /** {@inheritDoc} */
     @Override public void dispose() {
+        if (getSkinnable() == null) return;
```

Comment:
This will short-circuit the rest of the method, including the call to `super.dispose()`, if `getSkinnable()` is null. Are there any cases where this might miss some needed cleanup?

Reply:
that's intentional - the usual pattern to make dispose cope with repeated calls (something we learned in [JDK-8244112](https://bugs.openjdk.java.net/browse/JDK-8244112): the ony valid path to having skinnable null is a previous call to dispose (which is allowed by client code) in which we already did all clean-up.